internal/npm_install/generate_build_file.ts

```diff
@@ -194,11 +203,33 @@ js_library(
  * Generates all BUILD & bzl files for a package.
  */
 function generatePackageBuildFiles(pkg: Dep) {
-  // If a BUILD file was shipped with the package, append its contents to the end of
-  // what we generate for the package.
+  // If a BUILD file was shipped with the package we should symlink the generated BUILD file
+  // instead of append its contents to the end of the one we were going to generate.
+  // https://github.com/bazelbuild/rules_nodejs/issues/2131
   let buildFilePath: string|undefined;
   if (pkg._files.includes('BUILD')) buildFilePath = 'BUILD';
   if (pkg._files.includes('BUILD.bazel')) buildFilePath = 'BUILD.bazel';
+
+  // Recreate the pkg dir inside the node_modules folder
+  const nodeModulesPkgDir = `node_modules/${pkg._dir}`;
+  // Check if the current package dep dir is a symlink (which happens when we install a node_module
+  // with link:)
+  const isPkgDirASymlink =
+      fs.existsSync(nodeModulesPkgDir) && fs.lstatSync(nodeModulesPkgDir).isSymbolicLink();
+  // Check if the current package is also written inside the workspace
+  const isPkgInsideWorkspace = fs.realpathSync(nodeModulesPkgDir).includes(fs.realpathSync(`.`));
```

Comment:
Its a corner case, but if `symlink_node_modules = False` then yarn & npm are run in the root of the external repository. In that case, `link:` would be fairly useless as anything linked with a relative path would have to be copied over via that `data` attribute. Still, the comment here should reflect that `fs.realpathSync(`.`)` will not be the root of the workspace if `symlink_node_modules = False`

Reply:
@gregmagolan have you any suggestion to solve that corner case?

Reply:
Looking for approve from @gregmagolan to merge as-is